### ISSUE TITLE
Missing info (or typo?) in Readme.md

### ISSUE THREAD

**Issue by stone212** [postfixadmin/docker#7]
The README.MD file says this:

```
fpm-alpine

This image has a very small footprint. It is based on Alpine Linux and starts only a PHP FPM process. Use this variant if you already have a seperate webserver. If you need more tools, that are not available on Alpine Linux, use the fpm image instead.

fpm

This image starts only a PHP FPM container. Use this variant if you already have a seperate webserver.
```

See that both options say `Use this variant if you already have a seperate webserver.`  Probably one of them should say "if you *do not* already have a separate webserver"?

**Comment by DavidGoodwin** [comment]
Without looking, I'd assume that the 'fpm' variant is not as lightweight as the fpm-alpine; but otherwise they're probably nearly identical....

**Comment by J0WI** [comment]
> If you need more tools, that are not available on Alpine Linux, use the fpm image instead.

`fpm` is based on Debian. The functionality is identical.

**Comment by stone212** [comment]
@J0WI What do you mean "tools"?  What do these modules do?  There is no information in the doc.

**Comment by stone212** [comment]
@J0WI I remember installing this manually and it required a nightmare of custom mySQL tables.  How will the docker image interact with the accounts already in place?  Or does it contain mySQL inside the docker image?  Is that one of the tools you mean?

**Comment by DavidGoodwin** [comment]
tools -> linux commands/packages. None are probably needed unless you plan to have some sort of mailbox post creation script that does some magic.

The docker image should use/upgrade any existing database -> you can check by pointing your browser at http://localhost:whateverport/upgrade.php once you've got your docker image running.

**Comment by stone212** [comment]
@DavidGoodwin Thank you for replying but that is the most vague note I can imagine.

Where do you want me to choose `tools -> linux commands/packages` menu from?  Or are you saying that "tools means linux commands or packages"? and if that's the case then what tools are you talking about (my original question).

Then you say `The docker image should use/upgrade any existing database` but that is really vague.  How/where does it know to do that?  Will it break existing mySQL databases and tables that use a different structure?  When I set postfixadmin manually once I had to use completely non-standard table structure.

> you can check by pointing your browser at http://localhost:whateverport/upgrade.php once you've got your docker image running

Um.  What is `http://localhost:whateverport/upgrade.php`?  what port do you mean?  mySQL port?  Web port (default 80)?  And did I miss a doc that talks about "upgrade.php"?  

Your shortcuts make more confusion not less.

**Comment by DavidGoodwin** [comment]
Hi @stone212 - sorry - I'll try and reword things and make it easier to understand.

By Tools - What linux commands are available within the image. I strongly suspect the alpine-fpm based image would be sufficient for what you need - however it's possible you may hook scripts configured within postfixadmin (for example - after adding a mailbox a script may be executed to create a set folder structure or something) which would have a dependency on some external program that's not part of the alpine-fpm image.

When you run the docker image, you specify what database type/host/username/password to use through command line arguments. 

If you're running the container like : 

```
docker run -e POSTFIXADMIN_DB_TYPE=mysqli \
           -e POSTFIXADMIN_DB_HOST=some.host.name \
           -e POSTFIXADMIN_DB_USER=user \
           -e POSTFIXADMIN_DB_PASSWORD=topsecret \
           -e POSTFIXADMIN_DB_NAME=postfixadmin \
           --name postfixadmin \
           -p 8080:80 \
        postfixadmin-image
```

When the docker image runs, it executes php upgrade.php (see https://github.com/postfixadmin/docker/blob/master/docker-entrypoint.sh#L73 ) - which will check the database and install necessary schema changes.

If you have a non-standard database/schema, then I suspect you need to check what's in your database against what upgrade.php is likely to do.

**Comment by stone212** [comment]
@DavidGoodwin I'm starting to understand.  Trying now but held up by https://github.com/postfixadmin/docker/issues/8 so I will get back to this after I figure that out.

**Comment by stone212** [comment]
@DavidGoodwin Okay I got past that problem but now the Docker container has a couple of problems.  Maybe I should post new issues.  Here is what happens:

```
+ POSTFIXADMIN_DB_TYPE=mysqli
+ POSTFIXADMIN_DB_HOST=localhost
+ POSTFIXADMIN_DB_USER=username
+ POSTFIXADMIN_DB_PASSWORD=yoo7aaLaak3oNgiS
+ POSTFIXADMIN_SETUP_PASSWORD=
+ [[ php-fpm == apache2* ]]
+ '[' php-fpm == php-fpm ']'
+ '[' -e index.php -a -e scripts/postfixadmin-cli.php ']'
+ echo 'Postfixadmin not found in /var/www/html - copying now...'
Postfixadmin not found in /var/www/html - copying now...
++ ls -A
+ '[' '' ']'
+ tar xf -
+ tar cf - --one-file-system -C /usr/src/postfixadmin .
+ echo 'Complete! Postfixadmin has been successfully copied to /var/www/html'
Complete! Postfixadmin has been successfully copied to /var/www/html
+ case "${POSTFIXADMIN_DB_TYPE}" in
+ : 3306
+ '[' mysqli '!=' sqlite ']'
+ '[' -z username -o -z yoo7aaLaak3oNgiS ']'
+ timeout 15 bash -c 'until echo > /dev/tcp/localhost/3306; do sleep 0.5; done'
timeout: can't execute '15': No such file or directory
```

Two problems with this.  One is that `/var/www/html` still only contains the default apache2 `index.html` and the second problem is that I guess the script does not complete because of the timeout error (which I don't understand yet).

Anyway browsing the website shows an apache2 default page.  I will try a different variant.
